pkg/domain/domain.go

```diff
@@ -2367,6 +2370,27 @@ func (do *Domain) UpdateTableStatsLoop(ctx, initStatsCtx sessionctx.Context) err
 	return nil
 }
 
+// EnableStatsOwner enables this node to execute stats owner jobs.
+// Since ownerManager.CampaignOwner will start a new goroutine to run ownerManager.campaignLoop,
+// we should make sure that before invoking EnableStatsOwner(), stats owner is DISABLE.
+func (do *Domain) EnableStatsOwner() error {
+	if !do.statsOwner.IsOwner() {
+		err := do.statsOwner.CampaignOwner()
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// DisableStatsOwner disable this node to execute stats owner.
+// We should make sure that before invoking DisableStatsOwner(), stats owner is ENABLE.
+func (do *Domain) DisableStatsOwner() error {
+	if do.statsOwner.IsOwner() {
```

Comment:
I think this is not correct. After `EnableStatsOwner` this node have started the campaign. But it's not the owner. Then in `DisableStatsOwner` it will not be canceled and leaked.

Reply:
there are other problems, https://github.com/pingcap/tidb/issues/56053 I'll review the whole owner package later.

Reply:
this is a fundamental design issue; we can ask the infrastructure team to take a look.

Reply:
I see in DDL `func (d *ddl) DisableDDL()` we always cancel campaign no matter `IsOwner()`. Can you change to this way?

Reply:
> DisableDDL

I change to this way.